src/ape_ethereum/ecosystem.py

```diff
@@ -158,12 +160,45 @@ def decode(cls, data: dict) -> ReceiptAPI:
         )
 
 
+class BlockGasFee(BlockGasAPI):
```

Comment:
People can always iterate on this class as they see fit, I'd love to see some better block querying properties.

Reply:
Yeah, this needs more thought, you'd probably want to do some sort of query like:

```py
avg_base_fee = sum(chain.query_blocks(start=START_BLOCK, end=END_BLOCK).block.gas.base_fee) / (END_BLOCK - START_BLOCK)
```

Reply:
Or probably more pythonically

```py
avg_base_fee = sum(
    blk.gas.base_fee for blk in chain.query_blocks(start=START_BLOCK, end=END_BLOCK)
) / (END_BLOCK - START_BLOCK)
```

Reply:
This gets into https://github.com/ApeWorX/ape/issues/129

Reply:
In this querying use-case, they need to make sure the `START_BLOCK` post London fork. I think an exception getting raised when trying to query a base fee pre London makes sense. I don't think there is any concept of a base fee pre for block pre-London.

Reply:
We can look into handling that when it becomes a problem

Reply:
Yeah I think returning `None` is better again. The key is not there on the data pre London forks, so it is like the same thing, just safer: no `KeyErrors`.